File: app/src/main/java/com/worldventures/dreamtrips/wallet/domain/converter/WalletCoordinatesToSmartCardCoordinatesConverter.java
package com.worldventures.dreamtrips.wallet.domain.converter;

import com.worldventures.dreamtrips.api.smart_card.location.model.ImmutableSmartCardCoordinates;
import com.worldventures.dreamtrips.api.smart_card.location.model.SmartCardCoordinates;
import com.worldventures.dreamtrips.modules.mapping.converter.Converter;
import com.worldventures.dreamtrips.wallet.domain.entity.lostcard.WalletCoordinates;

import io.techery.mappery.MapperyContext;


public class WalletCoordinatesToSmartCardCoordinatesConverter implements Converter<WalletCoordinates, SmartCardCoordinates> {
   @Override
   public Class<WalletCoordinates> sourceClass() {
      return WalletCoordinates.class;
   }

   @Override
   public Class<SmartCardCoordinates> targetClass() {
      return SmartCardCoordinates.class;
   }

   @Override
   public SmartCardCoordinates convert(MapperyContext mapperyContext, WalletCoordinates walletCoordinates) {
      return ImmutableSmartCardCoordinates.builder()
            .lat(walletCoordinates.lat())
            .lng(walletCoordinates.lng())
            .build();
   }
}

File: wallet/src/main/java/com/worldventures/wallet/domain/converter/WalletCoordinatesToSmartCardCoordinatesConverter.kt
package com.worldventures.wallet.domain.converter

import com.worldventures.core.converter.Converter
import com.worldventures.dreamtrips.api.smart_card.location.model.ImmutableSmartCardCoordinates
import com.worldventures.dreamtrips.api.smart_card.location.model.SmartCardCoordinates
import com.worldventures.wallet.domain.entity.lostcard.WalletCoordinates

import io.techery.mappery.MapperyContext

class WalletCoordinatesToSmartCardCoordinatesConverter : Converter<WalletCoordinates, SmartCardCoordinates> {

   override fun sourceClass(): Class<WalletCoordinates> = WalletCoordinates::class.java

   override fun targetClass(): Class<SmartCardCoordinates> = SmartCardCoordinates::class.java

   override fun convert(context: MapperyContext, source: WalletCoordinates): SmartCardCoordinates {
      return ImmutableSmartCardCoordinates.builder()
            .lat(source.lat)
            .lng(source.lng)
            .build()
   }
}
